### ISSUE TITLE
`craft index-assets/all` console command only indexes first volume

### ISSUE THREAD

**Issue by benjaminkohl** [craftcms/cms#7741]
### Description

When I try to run the `php craft index-assets/all`console command, it only indexes the first volume. Additionally, when it attempts to index the other volumes, it reports that an indexed asset is missing its file and it always displays the _first volume's_ name in each attempt. Here is an obfuscated example of what I mean:

```
Indexing assets in VolumeONE ...
    > #0: file1.jpg ... done
    > #1: file2.jpg ... done
    > #2: file3.jpg ... done
    > #3: file4.jpg ... done
    > #4: file5.jpg ... done

Done indexing assets in VolumeONE.

One recorded asset is missing its file:
- VolumeONE/index.html (22)

Indexing assets in VolumeTWO ...
Done indexing assets in VolumeTWO.

One recorded asset is missing its file:
- VolumeONE/index.html (22)

Indexing assets in VolumeTHREE ...
Done indexing assets in VolumeTHREE.

One recorded asset is missing its file:
- VolumeONE/index.html (22)

Indexing assets in VolumeFOUR ...
Done indexing assets in VolumeFOUR.

One recorded asset is missing its file:
- VolumeONE/index.html (22)

```

If I run `php craft index-assets/one VolumeTWO`, it works as expected and I see it listing the files that it finds. This works for each individual asset volume.

I checked the console.log file and there are no recorded exceptions or errors of any kind.

### Steps to reproduce

1. Set up a Craft 3.6.8 - 3.6.11.2 environment that has multiple asset volumes that each contain files.
2. Run `php craft index-assets/all`

### Additional info

- Craft version: Started as 3.6.8 but I updated to 3.6.11.2 to see if that fixed it.
- PHP version: 7.3.27
- Database driver & version: MySQL 5.7.33
- Plugin Versions:

Cache Flag 1.2.4
CP Field Inspect 1.2.4
CP Filters 1.1.1
Craft Commerce 3.2.17.3
Feed Me 4.3.6
Imager X v3.3.0
Inventory 2.1.1
Link Vault 3.1.7
LJ Dynamic Fields 3.0.11
Minify 1.2.10
Redactor 2.8.5

**Comment by andris-sevcenko** [comment]
Thank you for reporting this - just pushed a fix for the next release!

**Comment by brandonkelly** [comment]
We just released Craft 3.6.12 with the fix for this.
